rgraphquery/queryresult.go

```diff
@@ -0,0 +1,108 @@
+package rgraphquery
+
+import (
+	"errors"
+	"github.com/gomodule/redigo/redis"
+)
+
+type QueryResult struct {
+	Rows [][]ResultCell
+	Headers []string
+}
+
+func createQueryResult(results interface{}) (*QueryResult, error) {
+	resultSet := &QueryResult{}
+
+	values, err := redis.Values(results, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if we've encountered an error
+	if err, ok := values[len(values) - 1].(redis.Error); ok {
+		return nil, err
+	}
+
+	if len(values) == 1 {
+		// Nothing
+	} else {
+		err := resultSet.parseValues(values)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Why not just
```
if len(values) > 1 {
  err := resultSet.parseValues(values)
  if err != nil {
    return nil, err
  }
}
```

And maybe a comment explaining that the first row is a headers row?